src/java.base/share/classes/java/lang/foreign/Linker.java

```diff
@@ -122,7 +123,34 @@
  * a pointer that is known to point to a C {@code int[2]} array can be modelled as an address layout whose
  * target layout is a sequence layout whose element count is 2, and whose element type is {@link ValueLayout#JAVA_INT}.
  * <p>
- * The following table shows some examples of how C types are modelled in Linux/x64:
+ * All native linker implementations are guaranteed to provide canonical layouts for the following set of types:
+ * <ul>
+ *     <li>{@code bool}</li>
+ *     <li>{@code char}</li>
+ *     <li>{@code short}</li>
+ *     <li>{@code int}</li>
+ *     <li>{@code long}</li>
+ *     <li>{@code long long}</li>
+ *     <li>{@code float}</li>
+ *     <li>{@code double}</li>
+ *     <li>{@code size_t}</li>
+ *     <li>{@code wchar_t}</li>
+ *     <li>{@code void*}</li>
+ * </ul>
+ * As noted above, the specific canonical layout associated with each type can vary, depending on the data model
+ * supported by a given ABI. For instance, the C type {@code long} maps to the layout constant {@link ValueLayout#JAVA_LONG}
+ * on Linux/x64, but maps to the layout constant {@link ValueLayout#JAVA_INT} on Windows/x64. Similarly, the C type
+ * {@code size_t} maps to the layout constant {@link ValueLayout#JAVA_LONG} on 64-bit platforms, but maps to the layout
+ * constant {@link ValueLayout#JAVA_INT} on 32-bit platforms.
+ * <p>
+ * A native linker typically does not provide canonical layouts for C's unsigned integral types. Instead, they are
+ * modelled using the canonical layouts associated with their corresponding signed integral types. For instance,
+ * the C type {@code unsigned long} maps to the layout constant {@link ValueLayout#JAVA_LONG} on Linux/x64, but maps to
+ * the layout constant {@link ValueLayout#JAVA_INT} on Windows/x64.
+ * <p>
+ * The following table shows some examples of how C types are modelled in Linux/x64 according to the
+ * "System V Application Binary Interface - AMD64 Architecture Processor Supplement" (all the examples provided
```

Comment:
I have seen some discussion on this and I agree an authoritative link does not exist (I have searched for it in the past). I guess I'm not super wild about also including "AMD64 Architecture Processor Supplement". E.g. IMHO "System V Application Binary Interface" is more than enough?

Reply:
Ok, I think I agree, given that we already name x64 as a platform.